src/app/dynamic-forms/fields/rich-text/rich-text.component.ts

```diff
@@ -310,6 +343,78 @@ export class RichTextComponent implements OnDestroy, AfterViewInit {
     }
   }
 
+  /**
+   * Insert Image 
+   */
+  
+  toggleImage(): void{
+    this.isTable = true;
+    // const url = window.prompt('URL')
+    // if (url) {
+    //   this.editor.chain().focus().setImage({ src: url }).run()
+    // }
+
+      const selection = this.editor.view.state.selection;
+      const selectedText = selection ? this.editor.state.doc.textBetween(selection.from, selection.to) : undefined;
+
+      const dialogRef: MatDialogRef<AddRichTextImageComponent> = this._dialog.open(
+        AddRichTextImageComponent,
+        {
+          data: {
+            content: selectedText
+          },
+          maxHeight: "80vh",
+          width: "600px"
+        }
+      );
+
+      dialogRef.afterClosed().subscribe({
+        next: (result: IRichTextLink) => {
+
+          if (result) {
+            this.editor
+              .chain()
+              .setImage({ src:  result.href })
```

Comment:
Have you tested behaviour of this with and without selected text? Does it replace selected text when used this way?